### DISCUSSION TITLE
Feature Request: Allow Deployment of Git Branch Other Than Main to a Development Environment

### DISCUSSION THREAD

**Discussion by paulsterling** [umbraco/Umbraco.Cloud.Issues#907] [Ideas]
### Issue description

Umbraco Cloud deployments use the "master" (or "main") branch when deploying to a Development environment. I propose allowing deployments from other Git branches. Additionally, I suggest providing an option to create a new environment directly from a specific Git branch (using the Flexible Environment feature). This flexibility will enable teams to test features, bug fixes, and other changes in isolated environments before merging them into the main branch.

I expect there are some additional considerations to be taken with this feature. Please feel free to add comments.

**Comment by skttl** [comment]
To change the deployment branch, you can do some Kudu magic, by sending a POST request to https://[your-project].scm.euwest01.umbraco.io/settings with the following JSON body

```json
{
    "key": "deployment_branch", 
    "value": "[the branch you want to deploy from]" 
}
```

The request needs basic authentication with your cloud user and password.

**Comment by Amalie-Wowern** [comment]
@skttl could this be added to the ci/cd setup? 
 Fx that test branch deploys developer enviroment and main deploys to staging

**Comment by paulsterling** [comment]
Thanks for the tip @skttl ! I'd still like to keep this feature request open since my thought is that it would be ideal to be able to do this from the portal when adding a new flexible environment to a project.
